app/src/main/java/framgia/com/ichat/data/reposiotries/LogInRepository.java

```diff
@@ -0,0 +1,40 @@
+package framgia.com.ichat.data.reposiotries;
+
+import android.content.SharedPreferences;
+
+import com.google.android.gms.tasks.Task;
+
+import framgia.com.ichat.data.source.CallBack;
+import framgia.com.ichat.data.source.login.LocalLogin;
+import framgia.com.ichat.data.source.login.LogInDataSource;
+import framgia.com.ichat.data.source.login.RemoteLogin;
+
+public class LogInRepository implements LogInDataSource.RemoteDataSource, LogInDataSource.LocalDataSource {
+    private SharedPreferences mShared;
+    private SharedPreferences.Editor mEditor;
+    private LocalLogin mLocalLogin;
+
+    public LogInRepository() {
+    }
+
+    public LogInRepository(SharedPreferences mShared, SharedPreferences.Editor mEditor) {
```

Comment:
```suggestion
    public LogInRepository(SharedPreferences sharedPreference) {
```

editor có thể khởi tạo bằng sharedPreference

Reply:
phần này e nên sử dụng AuthenicationDataSource trong pull request của Tuấn